### PR TITLE
dxSankey: add react/vue code snippets

### PR DESCRIPTION
**Pull request by Krijovnick**

### CODE REVIEW

File: concepts/05 Widgets/Sankey/05 Nodes and Links/05 Colorize Nodes and Links.md [hunk 1]
```diff
@@ -39,6 +39,51 @@ When the palette does not have enough colors to paint each node differently, it
...
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSankey
+            palette-extension-mode="alternate"
+            palette="Bright"
+            /* or a custom palette */
+            /* :palette="['#70c92f', '#f8ca00', '#bd1550']" */
+        >
+        </DxSankey>
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/05 Nodes and Links/05 Colorize Nodes and Links.md]: First of all, I suggest that you make this component self-closing - <dxSankey />. In this example, it doesn't contain any nested components. After that, let's make the same changes in this PR and in the [dxFunnel](https://github.com/DevExpress/devextreme-documentation/pull/973/files) snippets.

Then we should change the comments in the whole PR - these comments are made in the markup, so let's use the following style:
```
<DxSankey
    palette-extension-mode="alternate"
    palette="Bright"  />
    <!-- or a custom palette -->
    <!-- :palette="['#70c92f', '#f8ca00', '#bd1550']" -->

```

File: concepts/05 Widgets/Sankey/05 Nodes and Links/05 Colorize Nodes and Links.md [hunk 2]
```diff
@@ -39,6 +39,51 @@ When the palette does not have enough colors to paint each node differently, it
...
+    class App extends React.Component {
+        render() {
+            return (
+                <Sankey
+                    paletteExtensionMode="alternate"
+                    palette="Bright"
+                    /* or a custom palette */
+                    /* palette={['#70c92f', '#f8ca00', '#bd1550']} */
+                >
+                </Sankey>
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/05 Nodes and Links/05 Colorize Nodes and Links.md]: Let's make the same changes here. Make a self-closing component and write comments in JSX style:
```
<Sankey
    paletteExtensionMode="alternate"
    palette="Bright" />
{/* or a custom palette */}
{/* palette={['#70c92f', '#f8ca00', '#bd1550']} */}
```

File: concepts/05 Widgets/Sankey/05 Nodes and Links/05 Colorize Nodes and Links.md [hunk 3]
```diff
@@ -78,6 +119,44 @@ Links also support several [colorization modes](/api-reference/20%20Data%20Visua
         // ...
     })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSankey ... >
+            <DxLink color-mode="source" /> <!-- or "target" | "gradient" --!>
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/05 Nodes and Links/05 Colorize Nodes and Links.md]:
```suggestion
            <DxLink color-mode="source" /> <!-- or "target" | "gradient" -->
```

File: concepts/05 Widgets/Sankey/05 Nodes and Links/10 Hover/05 User Interaction.md [hunk 4]
```diff
@@ -87,4 +152,37 @@ Set [hoverEnabled](/api-reference/20%20Data%20Visualization%20Widgets/dxSankey/1
         // ...
     })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSankey :hoverEnabled="false" />
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/05 Nodes and Links/10 Hover/05 User Interaction.md]:
```suggestion
        <DxSankey :hover-enabled="false" />
```

File: concepts/05 Widgets/Sankey/05 Nodes and Links/10 Hover/10 Events.md [hunk 5]
```diff
@@ -58,6 +58,75 @@ The **Sankey** widget raises individual events for nodes and links when their ho
         // ...
     })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSankey
+            @nodeHoverChanged="onNodeHoverChanged"
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/05 Nodes and Links/10 Hover/10 Events.md]:
```suggestion
            @node-hover-changed="onNodeHoverChanged"
```

File: concepts/05 Widgets/Sankey/05 Nodes and Links/10 Hover/10 Events.md [hunk 6]
```diff
@@ -58,6 +58,75 @@ The **Sankey** widget raises individual events for nodes and links when their ho
...
+            @linkHoverChanged="onLinkHoverChanged"
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/05 Nodes and Links/10 Hover/10 Events.md]:
```suggestion
            @link-hover-changed="onLinkHoverChanged"
```

File: concepts/05 Widgets/Sankey/05 Nodes and Links/10 Hover/10 Events.md [hunk 7]
```diff
@@ -58,6 +58,75 @@ The **Sankey** widget raises individual events for nodes and links when their ho
...
+
+    <!-- tab: App.js -->
+    import React from 'react';
+    import Sankey from 'devextreme-react/sankey';
+
+    class App extends React.Component {
+        render() {
+            return (
+                <Sankey
+                    onNodeHoverChanged={onNodeHoverChanged}
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/05 Nodes and Links/10 Hover/10 Events.md]: I see that onNode/LinkHoverChanged are declared inside the class. Shouldn't we write like {this.onNodeHoverChanged}?

File: concepts/05 Widgets/Sankey/05 Nodes and Links/25 Access a Node or Link Using the API.md [hunk 8]
```diff
@@ -76,6 +130,49 @@ You can also access a node or link in the event handlers. For example, the [onNo
         // ...
     })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSankey @nodeClick="onNodeClick" />
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/05 Nodes and Links/25 Access a Node or Link Using the API.md]:
```suggestion
        <DxSankey @node-click="onNodeClick" />
```

File: concepts/05 Widgets/Sankey/15 Node Labels/05 Customize Labels.md [hunk 9]
```diff
@@ -56,6 +56,62 @@ In the following example, the **customizeText** function is used to add incoming
...
+
+    export default {
+        components: {
+            DxSankey,
+            DxLabel
+        },
+        methods: {
+            customizeText(node) {
+                return node.title + " (in: " + node.linksIn.reduce(weightsReducer, 0) + ", " +
+                    + "out: " + node.linksOut.reduce(weightsReducer, 0) + ")";
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/15 Node Labels/05 Customize Labels.md]: I examined the snippet but I don't understand why the previous line ends with **+**, and this line also begins with a **+**. Is it ok?

File: concepts/05 Widgets/Sankey/15 Node Labels/05 Customize Labels.md [hunk 10]
```diff
@@ -56,6 +56,62 @@ In the following example, the **customizeText** function is used to add incoming
...
+
+    const weightsReducer = (accumulator, currentValue) => {
+        return accumulator + currentValue.weight;
+    }
+
+    class App extends React.Component {
+        render() {
+            return (
+                <Sankey ... >
+                    <Label customizeText={customizeText} />
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/15 Node Labels/05 Customize Labels.md]:
```suggestion
                    <Label customizeText={this.customizeText} />
```

File: concepts/05 Widgets/Sankey/15 Node Labels/05 Customize Labels.md [hunk 11]
```diff
@@ -56,6 +56,62 @@ In the following example, the **customizeText** function is used to add incoming
...
+    <template> 
+        <DxSankey ... >
+            <DxLabel :customize-text="customizeText" />
+        </DxSankey>
+    </template>
+
+    <script>
+    import DxSankey, { DxLabel } from 'devextreme-vue/sankey';
+
+    const weightsReducer = (accumulator, currentValue) => {
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/15 Node Labels/05 Customize Labels.md:71]: Why `const` and not a method?

File: concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/01 User Interaction.md [hunk 12]
```diff
@@ -86,4 +130,50 @@ Change the **export**.[formats](/api-reference/20%20Data%20Visualization%20Widge
...
     })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSankey ... >
+            <DxExport
+                :enabled="true"
+                :formats="["PNG", "JPEG"]"
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/01 User Interaction.md]: let's move this array away from markup and declare it in the component <script> here and React.

File: concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/05 API.md [hunk 13]
```diff
@@ -33,6 +33,62 @@ To export the widget using the API, call the [exportTo(fileName, format)](/api-r
...
+
+    class App extends React.Component {
+        constructor(props) {
+            super(props);
+            this.sankeyRef = React.createRef();
+        }
+        render() {
+            return (
+                <Sankey ref={this.sankeyRef} >
+                </Sankey>
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/05 API.md]:
```suggestion
                <Sankey ref={this.sankeyRef} />
```

File: concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/05 API.md [hunk 14]
```diff
@@ -33,6 +33,62 @@ To export the widget using the API, call the [exportTo(fileName, format)](/api-r
         // ...
     })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSankey ref="sankey" >
+        </DxSankey>
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/05 API.md]:
```suggestion
        <DxSankey ref="sankey" />
```

File: concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/05 API.md [hunk 15]
```diff
@@ -87,4 +143,74 @@ You can also export several widgets simultaneously using their SVG markup. Call
     <dx-sankey id="sankeyContainer1" ... ></dx-sankey>
     <dx-sankey id="sankeyContainer2" ... ></dx-sankey>
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSankey ref="sankey1" ></DxSankey>
+        <DxSankey ref="sankey2" ></DxSankey>
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/05 API.md]:
```suggestion
        <DxSankey ref="sankey1" />
        <DxSankey ref="sankey2" />
```

File: concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/10 Events.md [hunk 16]
```diff
@@ -60,6 +60,67 @@ You can handle these events with functions. If the handling functions are not go
...
     })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSankey
+            @exporting="onExporting"
+            @exported="onExported"
+            @fileSaving="onFileSaving"
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/10 Events.md]:
```suggestion
            @file-saving="onFileSaving"
```

File: concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/10 Events.md [hunk 17]
```diff
@@ -60,6 +60,67 @@ You can handle these events with functions. If the handling functions are not go
...
+    import Sankey from 'devextreme-react/sankey';
+
+    class App extends React.Component {
+        render() {
+            return (
+                <Sankey
+                    onExporting={onExporting}
+                    onExported={onExported}
+                    onFileSaving={onFileSaving}
+                ></Sankey>
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/10 Events.md]:
```suggestion
                />
```

**Review by AlexJameson — CHANGES_REQUESTED** [review]

File: concepts/05 Widgets/Sankey/15 Node Labels/05 Customize Labels.md (hunk 9 above)
Comment (Krijovnick) [reply concepts/05 Widgets/Sankey/15 Node Labels/05 Customize Labels.md]: The previous line ends with ', '
Or you mean something else?

File: concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/05 API.md [hunk 18]
```diff
@@ -87,4 +141,74 @@ You can also export several widgets simultaneously using their SVG markup. Call
...
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template> 
+        <DxSankey ref="sankey1" />
+        <DxSankey ref="sankey2" />
+    </template>
+
+    <script>
+    import DxSankey from 'devextreme-vue/sankey';
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/05 API.md:153]:
```suggestion
    import DxSankey from "devextreme-vue/sankey";
```

File: concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/05 API.md [hunk 19]
```diff
@@ -87,4 +141,74 @@ You can also export several widgets simultaneously using their SVG markup. Call
...
+    import exportMethods from "devextreme/viz/export";
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/05 API.md]:
```suggestion
    import exportMethods from 'devextreme/viz/export';
```

File: concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/05 API.md [hunk 20]
```diff
@@ -87,4 +141,74 @@ You can also export several widgets simultaneously using their SVG markup. Call
...
+        }
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+    import Sankey from 'devextreme-react/sankey';
+    import exportMethods from "devextreme/viz/export";
```
Comment (AlexJameson) [inline concepts/05 Widgets/Sankey/99 Client-Side Exporting and Printing/05 API.md]:
```suggestion
    import exportMethods from 'devextreme/viz/export';
```

**Review by AlexJameson — APPROVED** [review]

**Pull request merged**